app/views/shared/_comment.html.erb

```diff
@@ -1 +1,6 @@
-<p><%= User.find(comment.user_id).name %>: <%= comment.text %> </p>
\ No newline at end of file
+<p><%= User.find(comment.user_id).name %>: <%= comment.text %> </p>
+<%if can? :destroy, comment%>
+  <button>
+    <%=link_to "Delete", comment, method: :delete%>
+  </button>
```

Comment:
- [ ] Currently, I am able to delete the comment of other users as well while my role is not admin. Please fix this.